ppa/archive/models.py

```diff
@@ -119,9 +133,60 @@ def populate_from_bibdata(self, bibdata):
             as instance of :class:`ppa.archive.hathi.HathiBibliographicRecord`
 
         '''
-        self.title = bibdata.title
+
+        # set fields from marc if available, since it has more details
+        if bibdata.marcxml:
+            # set title and subtitle from marc if possible
+            self.title = bibdata.marcxml['245']['a']
+
+            # according to PUL CAMS,
+            # 245 subfield contains the subtitle *if* the preceding field
+            # ends with a colon. (Otherwise could be a parallel title,
+            # e.g. title in another language).
+            # HOWEVER: metadata from Hathi doesn't seem to follow this
+            # pattern (possibly due to records being older?)
+
+            # subfields is a list of code, value, code, value
+            # iterate in paired steps of two starting with first and second
+            # for code, value in zip(bibdata.marcxml['245'].subfields[0::2],
+            #                        bibdata.marcxml['245'].subfields[1::2]):
+            #     if code == 'b':
+            #         break
+            #     preceding_character = value[-1:]
+
+            # if preceding_character == ':':
+            #     self.subtitle = bibdata.marcxml['245']['b'] or ''
+
+            # NOTE: skipping precebding character check for now
+            self.subtitle = bibdata.marcxml['245']['b'] or ''
+
+            # indicator 2 provides the number of characters to be
+            # skipped when sorting (could be 0)
+            try:
+                non_sort = int(bibdata.marcxml['245'].indicators[1])
+            except ValueError:
+                # at least one record has a space here instead of a number
+                # probably a data error, but handle it
+                # - assuming no non-sort characters
+                non_sort = 0
+
+            self.sort_title = bibdata.marcxml.title()[non_sort:]
+
+            self.author = bibdata.marcxml.author() or ''
+            if '260' in bibdata.marcxml:
```

Comment:
I think a note here for what `260` is in marc might be helpful down the line. Maybe.